node/src/components/consensus/highway_core/state.rs

```diff
@@ -566,8 +569,45 @@ impl<C: Context> State<C> {
                 return Err(VoteError::ValueAfterTerminalBlock);
             }
         }
-        // TODO: Validate against LNC.
-        Ok(())
+
+        match self.validate_lnc(wvote) {
+            None => Ok(()),
+            Some(vidx) => Err(VoteError::LncNaiveCitation(vidx)),
+        }
+    }
+
+    /// Validates whether `wvote` violates the LNC rule.
+    /// Returns index of the first equivocator that should have been endorsed but wasn't.
+    ///
+    /// Vote violates LNC rule if it cites naively an equivocation.
+    /// If it cites equivocator then it needs to endorse votes that cite equivocating votes.
+    fn validate_lnc(&self, wvote: &WireVote<C>) -> Option<ValidatorIndex> {
```

Comment:
Action items from meeting with

(1) @goral09 and @afck : rewrite this as a stub (always returning `None`).
(2) @goral09 will port the tests and we will `ignore` those tests that expect the LNC to fail